### PR TITLE
Added time and improved some Poketes and Attacks

### PR DESCRIPTION
**Pull request by lxgr-linux**
Added:
- [x] Ingame time
- [x] A clock to keep track of it
- [x] Time stops after 120s of no input to avoid afk-ing
- [x] Time depending NPCs
- [x] Time depending Poketes:
   - [x] Strcily night and day active Poketes
   - [x] Shown in the Dex
   - [x] Evolvepath corcos_day -> raupathor_day -> schmetterling
   - [x] Evolvepath corcos_night -> raupathor_night -> mothor
- [x] Update gen_wiki
- [x] Pokete-Care

### CODE REVIEW

**Comment by MaFeLP** [comment]
There is a bug that has to do with Pokete Care:

1. Putting a Pokete into care
2. Waiting for it to gain xp
3. Go the care and talk to the npc, answer `n` to the `do you want it back` question.
4. Go right by one and left by one (activating the npc dialog again)
5. Answer `y` to the `do you want it back` question.
6. Go to the pokete center and find out, you pokete has been duplicated.
7. Find out that the Deck will not close when either hitting `q` or `1` and that the python thread is hitting 100% your CPU performance.

**Comment by lxgr-linux** [comment]
Thanks for the info.
The cause for this behavior was that when `Figure.add_poke` was called and a clear spot in the deck was present, the method added the poke in the clear spot and at the end.
It's fixed now.

File: gen_wiki.py [hunk 1]
```diff
@@ -56,6 +56,16 @@ def overview(multi_page: bool = False) -> str:
           for i, name in enumerate(["poketes", "attacks", "types", "item",
                                     "effects"])) + "\n"
 
+    @staticmethod
+    def get_name(poke):
```
Comment (MaFeLP) [inline gen_wiki.py:60]: You missed using this method in line `218` when getting the name of the evolve pokete!

File: pokete_classes/timer.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+ ###
+  #
+ #""",
+]
+
+DOUBLE_POINT = """
+ ##
+
+ ##"""
+
```
Comment (MaFeLP) [inline pokete_classes/timer.py:70]:
```suggestion


```

File: pokete_classes/timer.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+class Time:
+    """Timer class to keep track of ingame time
```
Comment (MaFeLP) [inline pokete_classes/timer.py:72]:
```suggestion
    """Timer class to keep track of in-game time
```

File: pokete_classes/timer.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    ARGS:
+        init_time: The initial time used for the timer"""
+
+    def __init__(self, init_time=0):
+        self.time = init_time  # time in ingame minutes
```
Comment (MaFeLP) [inline pokete_classes/timer.py:77]:
```suggestion
        self.time = init_time  # time in in-game minutes
```

File: pokete_classes/timer.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        self.last_input = init_time
+
+    def formated(self):
```
Comment (MaFeLP) [inline pokete_classes/timer.py:80]:
```suggestion
    def formatted(self):
```

File: pokete_classes/timer.py [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        """Returns the ingame time in a formated manner"""
```
Comment (MaFeLP) [inline pokete_classes/timer.py:81]:
```suggestion
        """Returns the in-game time in a formatted manner"""
```

File: pokete_classes/timer.py [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        _t = self.normalized
+        hours = int(_t / 60)
+        minutes = _t % 60
+        return f"{hours:02}:{minutes:02}"
+
+    def emit_input(self):
+        """Sets the las tinput time to the current time"""
```
Comment (MaFeLP) [inline pokete_classes/timer.py:88]:
```suggestion
        """Sets the last input time to the current time"""
```

File: pokete_classes/timer.py [hunk 8]
```diff
@@ -0,0 +1,158 @@
...
+
+    def draw_letters(self, d_p=True, letter_obs=None):
+        """Method to draw the letters on the clock
+        ARGS:
+            d_p: Whether or not the DOUBLE_POINT should be shown
+            letter_obs: The letter objects of the former intervall"""
+        if letter_obs is None:
+            letter_obs = []
+        self.__rem_obs(letter_obs)
+        ftime = self.time.formated().replace(":", "")
```
Comment (MaFeLP) [inline pokete_classes/timer.py:142]: Method has been renamed above.

```suggestion
        ftime = self.time.formatted().replace(":", "")
```

File: pokete_classes/timer.py [hunk 9]
```diff
@@ -0,0 +1,158 @@
+"""Contains Classes to manage ingame time"""
```
Comment (MaFeLP) [inline pokete_classes/timer.py:1]:
```suggestion
"""Contains Classes to manage in-game time"""
```

File: pokete.py [hunk 10]
```diff
@@ -810,12 +845,12 @@ class ExtraActions:
...
-                    + [getattr(obmp.ob_maps["playmap_7"], i)
-                       for i in p_data.map_data["playmap_7"]["balls"] if
-                       "playmap_7." + i not in figure.used_npcs or
-                       not save_trainers]):
+        _map = obmp.ob_maps["playmap_7"]
+        for obj in _map.get_obj("inner_walls").obs\
+                   + [i.main_ob for i in _map.trainers]\
+                   + [obmp.ob_maps["playmap_7"].get_obj(i)
+                    for i in p_data.map_data["playmap_7"]["balls"] if
+                        "playmap_7." + i not in figure.used_npcs or not save_trainers]:
```
Comment (MaFeLP) [inline pokete.py:853]:
```suggestion
                + [i.main_ob for i in _map.trainers]\
                + [obmp.ob_maps["playmap_7"].get_obj(i)
                    for i in p_data.map_data["playmap_7"]["balls"] if
                    "playmap_7." + i not in figure.used_npcs
                    or not save_trainers]:
```

File: pokete.py [hunk 11]
```diff
@@ -1081,10 +1117,10 @@ def gen_obs():
...
     # NPCs
     for npc in npcs:
-        parse_obj(obmp.ob_maps[npcs[npc]["map"]], npc,
-                  NPC(npc, npcs[npc]["texts"], fn=npcs[npc]["fn"],
-                      chat=npcs[npc].get("chat", None)),
-                  npcs[npc])
+        _npc = npcs[npc]
+        NPC(npc, _npc["texts"], fn=_npc["fn"],
+            chat=_npc.get("chat", None)).add(obmp.ob_maps[_npc["map"]],
+            _npc["x"], _npc["y"])
```
Comment (MaFeLP) [inline pokete.py:1123]:
```suggestion
                                             _npc["x"], _npc["y"])
```

File: pokete.py [hunk 12]
```diff
@@ -1118,14 +1154,16 @@ def check_version(sinfo):
 Do you want to continue?", int(width * 2 / 3))):
             exiter()
 
-
```
Comment (MaFeLP) [inline pokete.py:1121]: There should be two empty lines, so no deletion of the line is necessary.

```suggestion

```

**Review by MaFeLP — CHANGES_REQUESTED** [review]
Nice work!

File: pokete_classes/event.py [hunk 13]
```diff
@@ -20,10 +21,15 @@ def set(self, event):
         ARGS:
             _ev: New char"""
         self._ev = event
+        self.emit_fn()
```
[the remark this answers is not in the document]
Comment (lxgr-linux) [reply pokete_classes/event.py:24]: But in this case the best way to handle this would be to throw an error, and that's actually what will happen.

**Comment by lxgr-linux** [comment]
> Nice work!

Thanks alot, I hope this will improve the game.

**Review by MaFeLP — APPROVED** [review]
Nice idea! Keep those coming!

**Pull request merged**